src/vuln_analysis/tools/filesystem_tools.py

```
@@ -0,0 +1,508 @@
+            return f"Error: {exc}"
+        if not resolved.exists():
+            return f"Error: path not found: {path}"
+        if not resolved.is_dir():
+            return f"Error: not a directory: {path}"
+        cap = self._max_glob_results
+        try:
+            matches = sorted(itertools.islice(
+                (str(p.relative_to(resolved)) for p in resolved.glob(pattern) if p.is_file()),
+                cap + 1,
+            ))
+        except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)